packages/salesforcedx-vscode-core/src/commands/templates/forceFunctionCreate.ts

```diff
@@ -22,47 +19,72 @@ import {
 import * as cp from 'child_process';
 import * as path from 'path';
 import * as vscode from 'vscode';
+import { channelService, OUTPUT_CHANNEL } from '../../channels';
 import { nls } from '../../messages';
 import { notificationService } from '../../notifications';
-import { sfdxCoreSettings } from '../../settings';
-import { telemetryService } from '../../telemetry';
+import { MetadataDictionary, MetadataInfo } from '../../util';
 import {
   CompositeParametersGatherer,
   SfdxCommandlet,
   SfdxWorkspaceChecker
 } from '../util';
-import { BaseTemplateCommand } from './baseTemplateCommand';
 import { FUNCTION_TYPE_JAVA, FUNCTION_TYPE_JS } from './metadataTypeConstants';
 
+import { generateFunction, Language } from '@heroku/functions-core';
+import { getRootWorkspacePath } from '../../util';
+
 const LANGUAGE_JAVA = 'java';
 const LANGUAGE_JAVASCRIPT = 'javascript';
-const LOG_NAME = 'force_function_create';
-export class ForceFunctionCreateExecutor extends BaseTemplateCommand {
+const LANGUAGE_TYPESCRIPT = 'typescript';
 
-  public build(data: FunctionInfo): Command {
-    if (data.language === LANGUAGE_JAVASCRIPT) {
-      this.metadata = FUNCTION_TYPE_JS;
-      this.setFileExtension('js');
-    } else if (data.language === LANGUAGE_JAVA) {
-      this.metadata = FUNCTION_TYPE_JAVA;
-      this.setFileExtension('java');
-    }
-    return new SfdxCommandBuilder()
-      .withDescription(nls.localize('force_function_create_text'))
-      .withArg('generate:function')
-      .withFlag('--name', data.fileName)
-      .withFlag('--language', data.language)
-      .withLogName(LOG_NAME)
-      .build();
+const LOG_NAME = 'force_function_create';
+export class ForceFunctionCreateExecutor extends LibraryCommandletExecutor<
+  any
+> {
+  constructor() {
+    super(nls.localize('force_function_create_text'), LOG_NAME, OUTPUT_CHANNEL);
   }
-
-  public runPostCommandTasks(data: FunctionInfo) {
-    const language = data.language;
+  public async run(response: ContinueResponse<FunctionInfo>): Promise<boolean> {
+    const { fileName, language } = response.data;
+    let metadata: MetadataInfo | undefined;
+    switch (language) {
+      case LANGUAGE_JAVASCRIPT:
+        metadata = MetadataDictionary.getInfo(FUNCTION_TYPE_JS);
+        metadata!.suffix = '.js';
```

Comment:
I don't think we need to use metadata!suffix. A `const fileExtension` or something seems cleaner to me. Thoughts?

Reply:
Agree! But, all the definitions in metadataDictionary (apexclass, triggers etc) all use suffix and this will be a much bigger refactor. I'm fine with that if you still want to, but can we make it a separate PR?